lib/osf-components/addon/components/addons-service/terms-of-service/component.ts

```diff
@@ -0,0 +1,152 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import IntlService from 'ember-intl/services/intl';
+
+import { AllProviderTypes} from 'ember-osf-web/packages/addons-service/provider';
+import ExternalStorageServiceModel, { TermsOfServiceCapabilities } from 'ember-osf-web/models/external-storage-service';
+import ExternalComputingServiceModel from 'ember-osf-web/models/external-computing-service';
+import ExternalCitationServiceModel from 'ember-osf-web/models/external-citation-service';
+
+interface Args {
+    provider: AllProviderTypes;
+}
+
+type CapabilityCategory =
+    TermsOfServiceCapabilities.ADD_UPDATE_FILES |
+    TermsOfServiceCapabilities.DELETE_FILES |
+    TermsOfServiceCapabilities.FORKING |
+    TermsOfServiceCapabilities.LOGS |
+    TermsOfServiceCapabilities.PERMISSIONS |
+    TermsOfServiceCapabilities.REGISTERING |
+    TermsOfServiceCapabilities.FILE_VERSIONS;
+
+type ServiceTranslationKey = 'storage' | 'computing' | 'citation';
+
+
+const capabilitiesToLabelKeyMap: Record<CapabilityCategory, string> = {
+    [TermsOfServiceCapabilities.ADD_UPDATE_FILES]: 'addons.terms.labels.add-update-files',
+    [TermsOfServiceCapabilities.DELETE_FILES]: 'addons.terms.labels.delete-files',
+    [TermsOfServiceCapabilities.FORKING]: 'addons.terms.labels.forking',
+    [TermsOfServiceCapabilities.LOGS]: 'addons.terms.labels.logs',
+    [TermsOfServiceCapabilities.PERMISSIONS]: 'addons.terms.labels.permissions',
+    [TermsOfServiceCapabilities.REGISTERING]: 'addons.terms.labels.registering',
+    [TermsOfServiceCapabilities.FILE_VERSIONS]: 'addons.terms.labels.file-versions',
+};
+
+const capabilitiesToTextKeyMap: Record<ServiceTranslationKey, Partial<Record<CapabilityCategory, any>>> = {
```

Comment:
Sorry about this typing... Also, I've kept the translation-keys decoupled from the actual `TermsOfServiceCapabilities` enum values in case any of those change

Reply:
That's quite the thing.